In [1]:
import pandas as pd

In [24]:
# loading the data dictionary
file_name = 'institution_data_dictionary.csv'
df_data_dict_inst = pd.read_csv(file_name)
df_data_dict_inst.shape

(3251, 11)

In [25]:
# loading the data dictionary
file_name = 'fieldofstudy_data_dictionary.csv'
df_data_dict_fos = pd.read_csv(file_name)
df_data_dict_fos.shape

(132, 10)

In [16]:
pd.set_option('display.max_rows', 30)
# Load the most recent data file with the instituion names
# file_inst = 'dataset/Most-Recent-Cohorts-Institution_03142022/Most-Recent-Cohorts-Institution.csv'
file_inst = 'dataset/CollegeScorecard_Raw_Data_03142022/Most-Recent-Cohorts-Institution.csv'
df_inst = pd.read_csv(file, low_memory=False)


In [17]:
df_inst.shape

(6694, 2989)

This above data shows that there are  schools with 2989 features to look at. 

In [18]:
# Load the most recent data file with the instituion names
# file_inst = 'dataset/Most-Recent-Cohorts-Institution_03142022/Most-Recent-Cohorts-Institution.csv'
file_field_of_study = 'dataset/Most-Recent-Cohorts-Field-of-Study_03142022/Most-Recent-Cohorts-Field-of-Study.csv'
df_field_of_study = pd.read_csv(file, low_memory=False)


In [19]:
df_field_of_study.shape

(6694, 2989)

In [21]:
lst_features_inst = ['INSTNM', 'CITY', 'STABBR', 'PREDDEG', 'ADM_RATE_ALL', 'ADM_RATE','SATVR25', 'SATVR75', 'SATMT25', 'SATMT75','SATWR25', 'SATWR75', 'UGDS']

lst_features_inst = ['INSTNM', 'CITY', 'STABBR', 'PREDDEG', 'ADM_RATE_ALL', 'ADM_RATE','SATVR25', 'SATVR75', 'SATMT25', 'SATMT75','SATWR25', 'SATWR75', 'UGDS', 'TUITIONFEE_IN', 'TUITIONFEE_OUT', 'C150_4', 'RET_FT4', 'C100_4', 'ICLEVEL', 'BOOKSUPPLY', 'ROOMBOARD_ON', 'OTHEREXPENSE_ON', 'ROOMBOARD_OFF', 'OTHEREXPENSE_OFF']

lst_features_inst = ['INSTNM', 'CITY', 'STABBR', 'PREDDEG', 'ADM_RATE_ALL', 'ADM_RATE','SATVR25', 'SATVR75', 'SATMT25', 'SATMT75','SATWR25', 'SATWR75', 'UGDS', 'TUITIONFEE_IN', 'TUITIONFEE_OUT', 'C150_4', 'RET_FT4', 'C100_4', 'ICLEVEL', 'BOOKSUPPLY', 'ROOMBOARD_ON', 'OTHEREXPENSE_ON', 'ROOMBOARD_OFF', 'OTHEREXPENSE_OFF']

In [22]:
df_features_inst = df_inst[lst_features_inst]
df_features_inst

INSTNM             CITY STABBR  \
0                     Alabama A & M University           Normal     AL   
1          University of Alabama at Birmingham       Birmingham     AL   
2                           Amridge University       Montgomery     AL   
3          University of Alabama in Huntsville       Huntsville     AL   
4                     Alabama State University       Montgomery     AL   
...                                        ...              ...    ...   
6689        Georgia Military College - Eastman          Eastman     GA   
6690  American College of Barbering - Florence         Florence     KY   
6691      HCI College - Fort Lauderdale Campus  Fort Lauderdale     FL   
6692          ABC Adult School - Cabrillo Lane         Cerritos     CA   
6693           Urban Barber College - San Jose         San Jose     CA   

      PREDDEG  LOCALE  ADM_RATE  SATVR75  SATMT75     UGDS  TUITIONFEE_IN  \
0           3    12.0    0.9175    510.0    500.0   5271.0        10024.0   
1           3    12.0    0.7366    680.0    685.0  13328.0         8568.0   
2           2    12.0       NaN      NaN      NaN    365.0         6950.0   
3           3    12.0    0.8257    720.0    770.0   7785.0        11122.0   
4           3    12.0    0.9690    510.0    510.0   3750.0        11068.0   
...       ...     ...       ...      ...      ...      ...            ...   
6689        0     NaN       NaN      NaN      NaN      NaN         6165.0   
6690        0     NaN       NaN      NaN      NaN      NaN            NaN   
6691        0     NaN       NaN      NaN      NaN      NaN        19404.0   
6692        0     NaN       NaN      NaN      NaN      NaN            NaN   
6693        0     NaN       NaN      NaN      NaN      NaN            NaN   

      TUITIONFEE_OUT  C150_4  RET_FT4  C100_4  ICLEVEL  BOOKSUPPLY  \
0            18634.0  0.2974   0.5689  0.0797        1      1600.0   
1            20400.0  0.6340   0.8310  0.4023        1      1200.0   
2             6950.0  0.3333      NaN  0.5000        1       900.0   
3            23518.0  0.5768   0.8291  0.2696        1      2120.0   
4            19396.0  0.3276   0.6127  0.1648        1      1600.0   
...              ...     ...      ...     ...      ...         ...   
6689             NaN     NaN      NaN     NaN        1         NaN   
6690             NaN     NaN      NaN     NaN        3         NaN   
6691             NaN     NaN      NaN     NaN        2         NaN   
6692             NaN     NaN      NaN     NaN        3         NaN   
6693             NaN     NaN      NaN     NaN        3         NaN   

      ROOMBOARD_ON  OTHEREXPENSE_ON  ROOMBOARD_OFF  OTHEREXPENSE_OFF  
0           9240.0           3090.0         9240.0            3090.0  
1          12307.0           5555.0        12307.0            5555.0  
2              NaN              NaN         9600.0            1600.0  
3          10400.0           3994.0        10400.0            3994.0  
4           6050.0           3392.0         7320.0            4228.0  
...            ...              ...            ...               ...  
6689           NaN              NaN            NaN               NaN  
6690           NaN              NaN            NaN               NaN  
6691           NaN              NaN            NaN               NaN  
6692           NaN              NaN            NaN               NaN  
6693           NaN              NaN            NaN               NaN  

[6694 rows x 20 columns]

In [ ]:
'''
ICLEVEL - indicates 4 year instituions which can include professional schools (Medical, Dental, Law) 
PREDDEG - indicates the predominant degree awarded, if it mostly awards graduate degrees it is better to leave it out because we are targeting our data for undergraduate programs.
'''

df_inst_4yr = df_inst_features.loc[(df_features['ICLEVEL'] == 1) & (df_features['PREDDEG'] == 3)]
df_4yr